src/bindings/python/flux/wrapper.py

```diff
@@ -310,8 +310,9 @@ def __getattr__(self, name):
 
     def __clear(self):
         # avoid recursion
-        handle = self._handle
-        if handle is None:
+        if "_handle" in dir(self) and self._handle is not None:
```

Comment:
If you `raise AttributeError` when `name == lib` as suggested above, then the `"_handle" in dir(self)` here can be changed to `hasattr(self, "_handle")`.  They will both have the same outcome for right now, but if we ever overload `__dir__`, `dir(self)` could break too.  It's a nitpick though, so if you don't have the time, just let me know and we can do it in a separate PR.

Reply:
Done!